google-cloud-core/src/main/java/com/google/cloud/StringEnumValue.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud;
+
+import com.google.api.core.InternalApi;
+import com.google.common.base.Preconditions;
+
+/**
+ * This represents a specific instance of a concept having a known set of acceptable String values,
+ * which can expand later due to new API features. Standard Java enums can't be used in such
+ * a context.
+ */
+public abstract class StringEnumValue {
+  private String constant;
+
+  @InternalApi
```

Comment:
`StringEnumValue` itself isn't internal - public classes extend it. I just don't want external code creating subclasses.